### PR TITLE
Ca config collections - showing as outdated reference when quick publishing

### PR DESCRIPTION
**Pull request by srikanthgurram-17**
This is to solve https://github.com/wcm-io/io.wcm.caconfig.editor/issues/48

Below is the behaviour of nested configurations :

Current behaviour: When we update or add one config in nested config all the configs inside the nested configs last modified dates are updated.
Expect behaviour: When we update or add a single config in nested config only the config that is added or updated should have last modified date updated

### CODE REVIEW

**Comment by srikanthgurram-17** [comment]
@stefanseifert  please review!

**Comment by srikanthgurram-17** [comment]
@stefanseifert  gentle reminder, As I am away after Friday until next year to do any changes.

File: src/main/java/io/wcm/caconfig/extensions/persistence/impl/PersistenceUtils.java [hunk 1]
```diff
@@ -281,6 +282,34 @@ public static void commit(ResourceResolver resourceResolver, String relatedResou
...
+  public static boolean isItemModifiedOrNewlyAdded(ResourceResolver resolver, String resourcePath, ConfigurationPersistData item) {
+    Resource resource = resolver.getResource(resourcePath);
+    if (resource == null) {
+      return true; // Resource does not exist, so it is considered modified
+    }
+
+    ValueMap currentProperties = resource.getValueMap();
+    Map<String, Object> itemProperties = item.getProperties();
+
+    for (Map.Entry<String, Object> entry : itemProperties.entrySet()) {
```
Comment (stefanseifert) [inline src/main/java/io/wcm/caconfig/extensions/persistence/impl/PersistenceUtils.java:302]: this check ignores one edge case: assume a property was removed from the new properties set - in that case the property has to be removed from the existing resource as well, so this changes goes unnoticed by this method. so you need to check also the set of property names of the existing resource and the new properties list, filtering out the ignored properties as it is done in the `replaceProperties` method.

this edge case should also be covered in the unit test.

**Review by stefanseifert — CHANGES_REQUESTED** [review]
thanks for the contribution!

looks very good overall, i have found one edge case as open issue.

File: src/main/java/io/wcm/caconfig/extensions/persistence/impl/PersistenceUtils.java (hunk 1 above)
Comment (stefanseifert) [reply src/main/java/io/wcm/caconfig/extensions/persistence/impl/PersistenceUtils.java:302]: alternatively, i think you can compare two maps directly using their equals method - but make sure to use filtered maps removing the ignored properties. still, the edge cases should be covered by unit test.

Comment (srikanthgurram-17) [reply src/main/java/io/wcm/caconfig/extensions/persistence/impl/PersistenceUtils.java]: Updated the code with the edge case covered now Thanks

**Review by stefanseifert — APPROVED** [review]
